src/Mod/TechDraw/Gui/TaskHatch.cpp

```diff
@@ -186,7 +186,8 @@ void TaskHatch::apply(bool forceUpdate)
     if (m_dvp) {
         //only need requestPaint to hatch the face
         //need a recompute in order to claimChildren in tree
-        m_dvp->recomputeFeature();
+        m_dvp->touch();
+        m_dvp->getDocument()->recompute();
```

Comment:
I have no experience with the recompute API. What does this mean? That we should recompute the whole document rather than the View?

Reply:
Actually it's the touch that is solving the claim issue. So 
```
        m_dvp->touch();
        m_dvp->recomputeFeature();
```

Works to fix the claim issue.

But like this, for some reason the hatch object is then marked to recompute (the blue mark). 
Which is why I used this :

```
        m_dvp->touch();
        m_dvp->getDocument()->recompute();
```
Which is also what the geometric hatch is using (see in it's command activated function).

If I am not mistaken, the doc->recompute only recomputes what is marked to recompute. ie what is touched. So in this case the m_dvp. Which somehow makes the hatch be touched too. So it's also recomputed.

Reply:
touch marks m_dvp as needing to be recomputed.  the next document recompute will update m_dvp and every object linked to it.  recomputeFeature will only update m_dvp.

I've never been able to figure out what triggers the ViewProvider to run claimChildren.